app/server/appsmith-server/src/main/java/com/appsmith/server/services/AnalyticsService.java

```diff
@@ -65,6 +63,16 @@ public void sendEvent(String event, String userId, Map<String, Object> propertie
             return;
         }
 
+        if (!commonConfig.isCloudHosted()) {
+            userId = DigestUtils.sha256Hex(userId);
+            if (properties.containsKey("username")) {
+                properties.put("username", userId);
```

Comment:
```suggestion
            if (properties.containsKey("username")) {
                userId = DigestUtils.sha256Hex(userId);
                properties.put("username", userId);
```
Super minor: This ensures that we aren't hashing the userId for all events. Only those events which have the userId will undergo any hashing.

Reply:
Sorry, my bad. The `userId` is being used in other places in this function. I didn't notice that.